site/content/en/docs/Guides/Client SDKs/csharp.md

```diff
@@ -201,6 +201,155 @@ var playerId = "player1";
 bool isConnected = await agones.Alpha().IsPlayerConnectedAsync(playerId);
 ```
 
+## Counters
+
+### Alpha: GetCounterCount
+
+Returns the Count for a Counter, given the Counter's key (name). Will error if the key was not
+predefined in the GameServer resource on creation.
+
+```csharp
+string key = "rooms";
+long count = await agones.Alpha().GetCounterCountAsync(key);
+```
+
+### Alpha: SetCounterCount
+
+Sets a count to the given value. Use with care, as this will overwrite any previous invocations’ value.
+Cannot be greater than Capacity.
+
+```csharp
+string key = "rooms";
+long amount = 0;
+await agones.Alpha().SetCounterCountAsync(key, amount);
+```
+
+### Alpha: IncrementCounter
+
+Increases a counter by the given nonnegative integer amount. Will execute the increment operation
+against the current CRD value. Will max at max(int64). Will error if the key was not predefined in
+the GameServer resource on creation. Errors if the count is at the current capacity (to the latest
+knowledge of the SDK), and no increment will occur.
+
+Note: A potential race condition here is that if count values are set from both the SDK and through
+the K8s API (Allocation or otherwise), since the SDK append operation back to the CRD value is
+batched asynchronous any value incremented past the capacity will be silently truncated.
+
+```csharp
+string key = "rooms";
+long amount = 1;
+await agones.Alpha().IncrementCounterAsync(key, amount);
+```
+
+### Alpha: DecrementCounter
+
+Decreases the current count by the given nonnegative integer amount. The Counter Will not go below 0.
+Will execute the decrement operation against the current CRD value. Errors if the count is at 0 (to
+the latest knowledge of the SDK), and no decrement will occur.
+
+```csharp
+string key = "rooms";
+long amount = 2;
+await agones.Alpha().DecrementCounterAsync(key, amount);
+```
+
+### Alpha: SetCounterCapacity
+
+Sets the capacity for the given Counter. A capacity of 0 is no capacity.
+
+```csharp
+string key = "rooms";
+long amount = 0;
+await agones.Alpha().SetCounterCapacityAsync(key, amount);
+```
+
+### Alpha: GetCounterCapacity
+
+Returns the Capacity for a Counter, given the Counter's key (name). Will error if the key was not
+predefined in the GameServer resource on creation.
+
+```csharp
+string key = "rooms";
+long count = await agones.Alpha().GetCounterCapacityAsync(key);
+```
+## Lists
+
+### Alpha: AppendListValue
+
+Appends a string to a List's values list, given the List's key (name) and the string value. Will

```

Comment:
Will also error if it goes over capacity?

Reply:
Yep, yes it does.